The latest 3.9.0 release is breaking the WASI build

<!-- Thanks for contributing to Swift Crypto! Before you submit your issue, please make sure you followed our checklist and check the appropriate boxes by putting an x in the [ ]: [x] -->

### New Regression Checklist

- [x] I'm using the latest version of Swift Crypto (main branch)
- [x] I read the [Contribution Guidelines](https://github.com/apple/swift-crypto/blob/main/CONTRIBUTING.md)
- [x] I searched for [existing GitHub issues](https://github.com/apple/swift-crypto/issues)

### Regression Information
<!-- Knowing the breaking versions and last working versions helps us track down the regression easier -->
- Breaking version: 3.9.0
- Last working version: 3.8.1

### Regression Description

I'm using the following dependency version here. Since the dependency is optional and controlled by ENV(Disable by default on macOS platform), the Package.resolved does not contains a fix version of swift-crypto. So the WASI build (enabled swift-crypto dependency) will fetched the latest `3.9.0` version here.

```
.package(url: "https://github.com/apple/swift-crypto.git", from: "3.8.0")
```

The error message is as following.

```
/<redacted>/.build/checkouts/swift-crypto/Sources/CryptoBoringWrapper/Util/RandomBytes.swift:38:37: error: cannot find 'rng' in scope
        var remainingWord: UInt64 = rng.next()
```

### Swift Crypto version/commit hash

_[the Swift Crypto tag/commit hash]_

### Environment

Stand in the root of the repo and run `./scripts/environment.sh` and paste the output below

<details>
  <pre>PLEASE_REPLACE_THIS_STIRNG_WITH_OUTPUT_OF_script_environments_THAT_IS_IN_YOUR_PASTEBOARD</pre>
</details>

--- a/Sources/CryptoBoringWrapper/Util/RandomBytes.swift
+++ b/Sources/CryptoBoringWrapper/Util/RandomBytes.swift
@@ -21,9 +21,6 @@ extension UnsafeMutableRawBufferPointer {
 
         #if canImport(Darwin) || os(Linux) || os(Android) || os(Windows)
         var rng = SystemRandomNumberGenerator()
-        #else
-        fatalError("No secure random number generator on this platform.")
-        #endif
         precondition(count <= self.count)
 
         // We store bytes 64-bits at a time until we can't anymore.
@@ -41,6 +38,9 @@ extension UnsafeMutableRawBufferPointer {
             remainingWord >>= 8
             targetPtr = UnsafeMutableRawBufferPointer(rebasing: targetPtr[1...])
         }
+        #else
+        fatalError("No secure random number generator on this platform.")
+        #endif
     }
 }
 

